### PR TITLE
Third problem solution

### PR DESCRIPTION
**Pull request by giftnichakul**

### CODE REVIEW

File: solution/DiverseWord.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                ans += "c"
+                con_c += 1
+                c-=1
+                con_a = 0
+                con_b = 0
+            else
+                break
+            end
+        end
+        return ans
```
Comment (WatcharitNew) [inline solution/DiverseWord.rb]: No need for word return on last line here

File: spec/TestDiverseWord.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        return false if word[i] == word[i+1] && word[i+1] == word[i+2]
+    end
+
+    return true
+end
+
+RSpec.describe 'example test' do
+    it "example1" do
+        test = DiverseWord.new(6,1,1).solution
+        expect(checkResult(test, 6,1,1)).to eq true
```
Comment (WatcharitNew) [inline spec/TestDiverseWord.rb]: for checking boolean, we prefer `to be true` or  `to be false`

File: solution/diverse_word.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  def solution
+    ans = ''
+    a = @a
+    b = @b
+    c = @c
+    con_a = 0
+    con_b = 0
+    con_c = 0
+    while a != 0 || b != 0 || c != 0
+      if (a >= b && a >= c && con_a < 2) || (con_b == 2 && a >= 1) || (con_c == 2 && a >= 1)
```
Comment (komsun-k) [inline solution/diverse_word.rb]: Leave a line break before block to improve readability.
```suggestion
    con_b = 0
    con_c = 0

    while a != 0 || b != 0 || c != 0
      if (a >= b && a >= c && con_a < 2) || (con_b == 2 && a >= 1) || (con_c == 2 && a >= 1)
```

File: spec/test_diverse_word.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      expect(checkResult(test, 1, 3, 1)).to be true
+    end
+
+    it 'example3' do
+      test = DiverseWord.new(0, 1, 8).solution
+      expect(checkResult(test, 0, 1, 8)).to be true
+    end
+  end
+  describe 'Correctness test' do
+    it 'super_small' do
```
Comment (komsun-k) [inline spec/test_diverse_word.rb]: Also leave a line break here.
```suggestion
    end
  end

  describe 'Correctness test' do
    it 'super_small' do
```

**Review by komsun-k — APPROVED** [review]
Just need to improve the readability as my comments.

**Review by kanatsanan6 — APPROVED** [review]

**Pull request merged**